IntroOOP/RationalNumbers/RatNum.cs

```diff
@@ -0,0 +1,369 @@
+﻿namespace IntroOOP.RationalNumbers;
+
+
+public class RatNum
+{
+    private int _Numerator;     // Числитель
+    private int _Denominator;   // Знаменатель
+
+    public int Numerator
+    {
+        get => _Numerator;
+        set
+        {
+            if (value >= 0)
+                _Numerator = value;
+            else
+                throw new InvalidDataException("Числитель должен быть неотрицательным!");
+        }
+    }
+
+    public int Denominator
+    {
+        get => _Denominator;
+        set
+        {
+            if (value > 0)
+                _Denominator = value;
+            else
+                throw new InvalidDataException("Знаменатель должен быть натуральным числом!");
+        }
+    }
+
+    public RatNum()
+    {
+        Numerator = 1;
+        Denominator = 1;
+    }
+    public RatNum(int numerator, int denominator) : this(numerator)
+    {
+        Numerator = numerator;
+        Denominator = denominator;
+    }
+
+    // Если указан только числитель, то знаменателю устанавливается значение 1.
+    public RatNum(int numerator)
+    {
+        Numerator = numerator;
+        Denominator = 1;
+    }
+
+    // Возвращает Наибольший общий делитель (Алгоритм Евклида)
+    private static int GetCommonDivisor(int a, int b)
+    {
+        while (b != 0)
+        {
+            int temp = b;
+            b = a % b;
+            a = temp;
```

Comment:
С недавних пор можно написать красиво с использованием кортежей и деконструкции. Но ваш вариант будет работать в общем случае быстрее. Красота - не всегда оптимально по скорости и памяти!
```suggestion
            (a, b) = (b, a % b);
```